community/kernel/src/main/java/org/neo4j/kernel/impl/store/PropertyStore.java

```diff
@@ -364,12 +364,20 @@ public PropertyRecord getRecord( PropertyRecord record )
     public PageCursor newReadCursor( long recordId ) throws IOException
     {
         PageCursor cursor = storeFile.io( pageIdForRecord( recordId ), PF_SHARED_LOCK );
-        if ( !cursor.next() )
+        try
+        {
+            if ( !cursor.next() )
+            {
+                throw new IOException( "Record not found: " + recordId );
+            }
+            cursor.setOffset( (int) (recordId * RECORD_SIZE % storeFile.pageSize()) );
+            return cursor;
+        }
+        catch ( Throwable failure )
         {
-            throw new IOException( "Record not found: " + recordId );
+            cursor.close();
+            throw failure;
```

Comment:
These changes make the acquisition of the property page cursor more resilient.
Technically unrelated, just something I found while looking around this code.